david_laster/midterm/main.rb

```diff
@@ -0,0 +1,114 @@
+require_relative 'lib/animal'
+require_relative 'lib/client'
+require_relative 'lib/shelter'
+require_relative 'lib/seeds'
+
+
+
+
+
+
+menu = "
+888    888                            d8b 88888888888       d8b 888          
+888    888                            Y8P     888           Y8P 888          
+888    888                                    888               888          
+8888888888  8888b.  88888b.  88888b.  888     888   8888b.  888 888 .d8888b  
+888    888     \"88b 888 \"88b 888 \"88b 888     888      \"88b 888 888 88K      
+888    888 .d888888 888  888 888  888 888     888  .d888888 888 888 \"Y8888b. 
+888    888 888  888 888 d88P 888 d88P 888     888  888  888 888 888      X88 
+888    888 \"Y888888 88888P\"  88888P\"  888     888  \"Y888888 888 888  88888P' 
+                    888      888                                             
+                    888      888                                             
+                    888      888                                             
+
+Create [A]nimal
+Create [C]lient
+List [AA]nimals
+List [CC]lients
+[-] Adopt
+[+] Put Up For Adoption
+[Q]uit
+Choose Your Selection: "
+
+
+def new_animal(a)
+  puts "Enter the Animal's Name:"
+  name = gets.chomp
+
+  puts "Enter the Animal's Species:"
+  species = gets.chomp
+  a.animals << Animal.new(name,species)
+end
+
+def new_client(c)
+  puts "Enter the Client's name:"
+  name = gets.chomp
+
+  puts "Enter the Client's age:"
+  age = gets.chomp.to_i
+  c.clients << Client.new(name,age)
+end
+
+
+def adopt(shelter)
+  shelter.animals.each_with_index do |animal, index|
+    puts "#{index} #{animal.name}"
+  end
+  puts "Choose animal number to be adopted:"
+  animal_number = gets.chomp.to_i
+  shelter.clients.each_with_index do |client, index|
+    puts "#{index} #{client.name}"
+  end
+  puts "Choose client number to receive pet:"
+  client_number = gets.chomp.to_i
+
+  shelter.clients[client_number].pets << shelter.animals[animal_number]
+  shelter.animals[animal_number] = nil
```

Comment:
I understand your logic here - but this is going to throw an error the next time you try and print the shelter's animals. After you set one of the objects to nil it will look like this: `shelter.animals` will return

 `[#<Animal:0x007fde498320a0 @name="Kitchka", @species="Tiger Cat", @toys=[]>, nil, #<Animal:0x007fde49831b00 @name="Sassy", @species="Orange Cat", @toys=[]>`

and when you try and loop over the array like this
```
shelter.animals.each_with_index do |animal, index|
    puts "#{index} #{animal.name}"
end
```
it will throw an error and say it can't call `.name` on a nil object. This will happen when you try and adopt 2 animals.

What you need to do instead is use the `delete_at` method on line 65 so that you can remove the object all together instead of replacing it will nil. Like so:
```
 shelter.clients[client_number].pets << shelter.animals.delete_at(animal_number)
```
and you can get rid of line 66 all together.

give it a try!